src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1660,7 +1668,24 @@ private static HttpClient CreateHttpClient(HttpClientParameters parameters, Http
 
                     try
                     {
-                        socket.NoDelay = true;
+                        if (parameters.ServicePoint is not null)
```

Comment:
In what situation will the ServicePoint be null?

Reply:
If nobody explicitly accesses to `ServicePoint` property, in case underlying `ServicePoint` field will be null, but we can always pass the property itself and forces it to find necessary `ServicePoint`.